app/controllers/students/tests_controller.rb

```diff
@@ -1,8 +1,17 @@
 class Students::TestsController < ApplicationController
 
+  def index
+    @test_setting = TestSetting.find(params[:test_setting_id])
+    @questions = @test_setting.questions
+    tests_belongs_to_test_setting = Test.where(test_setting_id: @test_setting.id)
+    @attempts_number = tests_belongs_to_test_setting.where(user_id: current_user.id)
+    @test = current_user.tests.last
```

Comment:
I don't know another way how to assign smth to @test variable. User is redirected to this page after test passing.